From 6a2fbef4d86a1c4a9d0d333c41890ed9d89dc4b0 Mon Sep 17 00:00:00 2001
From: Tejas Patel <tejas.patel@xilinx.com>
Date: Thu, 12 Mar 2020 07:40:47 -0700
Subject: [PATCH 1268/1851] clk: zynqmp: Fix CLK_FRAC bit index

commit e238d51d2ded214cd6f62c38fe6c6bdab4363f03 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes fdbb2e62f6a5 (clk: zynqmp: Update CLK_FRAC support as per upstream)

CLK_FRAC is stored in BIT(4) of in TF-A type flags. While storing
custom type flags in custom_type_flag variable it is already being
shifted by 4. So in custom_type_flag variable bit index of CLK_FRAC
flag would be 0 and not 4.

Signed-off-by: Tejas Patel <tejas.patel@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/clk/zynqmp/divider.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/zynqmp/divider.c b/drivers/clk/zynqmp/divider.c
index 7613678cdc89..3cadc4744788 100644
--- a/drivers/clk/zynqmp/divider.c
+++ b/drivers/clk/zynqmp/divider.c
@@ -26,7 +26,7 @@
 	container_of(_hw, struct zynqmp_clk_divider, hw)
 
 #define CLK_FRAC		BIT(13) /* has a fractional parent */
-#define CUSTOM_FLAG_CLK_FRAC	BIT(4) /* has a fractional parent in custom type flag */
+#define CUSTOM_FLAG_CLK_FRAC	BIT(0) /* has a fractional parent in custom type flag */
 
 /**
  * struct zynqmp_clk_divider - adjustable divider clock
-- 
2.31.1

